src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,76 @@
-export const Pagination = () => {};
+import React from 'react';
+import classNames from 'classnames';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (pageNumber: number) => void
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pages = getNumbers(1, Math.ceil(total / perPage));
+
+  return (
+    <ul className="pagination">
+      <li className={classNames(
+        'page-item',
+        { disabled: currentPage === 1 },
+      )}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={() => (currentPage !== 1
+            ? onPageChange(currentPage - 1)
+            : null)}
```

Comment:
As I have understood this function will be created every time when will be happened render if we don't create a method
But maybe I`m wrong)

Reply:
when we rerender it will dissapear, and on new object we will have new callback
in general it works like your's when you check 
`const getPrevPage = () => {
    if (currentPage !== 1) {
      onPageChange(currentPage - 1);
    }
  };`
  
 maybe in your case you will have undefined